boto/cloudfront/invalidation.py

```diff
@@ -68,10 +68,11 @@ def __setitem__(self, k, v):
         self.paths[k] = v
 
     def escape(self, p):
-        """Escape a path, make sure it begins with a slash and contains no invalid characters"""
+        """Escape a path, make sure it begins with a slash and contains no invalid characters. Retain literal wildcard characters."""
         if not p[0] == "/":
             p = "/%s" % p
-        return urllib.parse.quote(p)
+        p_quoted = urllib.parse.quote(p)
+        return p_quoted.replace('%2A', '*')
```

Comment:
I would prefer to make the wildcard a safe character via the quote method: https://docs.python.org/2/library/urllib.html#urllib.quote

A little safer than replacing all ``%2A``.